File: cs/ParkingTest/ParkingManagerTest.cs
using Parking;
using System.Collections.Generic;
using Xunit;

namespace ParkingTest
{
    public class ParkingManagerTest
    {
        readonly ParkingManager parkingManager;
        readonly IReadOnlyList<ParkingLot> parkingLots;

        public ParkingManagerTest()
        {
            var parkingLot1 = new ParkingLot(1);
            var parkingLot2 = new ParkingLot(1);
            parkingLots = new List<ParkingLot> { parkingLot1, parkingLot2 };
            parkingManager = new ParkingManager(new List<ParkingBoy> {
                new ParkingBoy(new List<ParkingLot> { parkingLot1 }, new OderParkingBoyRule()),
                new ParkingBoy(new List<ParkingLot> { parkingLot2 }, new MostAvailableParkingBoyRule())
            });
        }

        [Fact]
        public void Should_return_one_of_the_parking_lots()
        {
            var parkingLot = parkingManager.FindOneValidParkingLot();

            Assert.Contains(parkingLot, parkingLots);
        }

        [Fact]
        public void Should_return_another_one_when_parked_one_car()
        {
            var oneLot = parkingManager.FindOneValidParkingLot();
            oneLot.Park(new Car("川A 11111"));

            var anotherLot = parkingManager.FindOneValidParkingLot();

            Assert.Contains(anotherLot, parkingLots);
            Assert.NotEqual(oneLot, anotherLot);
        }
    }
}

File: kt/src/test/kotlin/com/github/scglwsj/parking/ParkingManagerTest.kt
package com.github.scglwsj.parking

import com.github.scglwsj.parking.boy.MostAvailableParkingBoyRule
import com.github.scglwsj.parking.boy.OderParkingBoyRule
import com.github.scglwsj.parking.boy.ParkingBoy
import com.github.scglwsj.parking.lot.Car
import com.github.scglwsj.parking.lot.ParkingLot
import com.github.scglwsj.parking.manager.ParkingManager
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class ParkingManagerTest {
    private lateinit var parkingManager: ParkingManager
    private lateinit var parkingLots: List<ParkingLot>

    @BeforeEach
    fun setUp() {
        val parkingLot1 = ParkingLot(1)
        val parkingLot2 = ParkingLot(1)
        parkingLots = listOf(parkingLot1, parkingLot2)
        parkingManager = ParkingManager(
            listOf(
                ParkingBoy(listOf(parkingLot1), OderParkingBoyRule()),
                ParkingBoy(listOf(parkingLot2), MostAvailableParkingBoyRule())
            )
        )
    }

    @Test
    fun `should return one of the parking lots`() {
        val parkingLot = parkingManager.findOneValidParkingLot()!!

        assertTrue(parkingLot in parkingLots)
    }

    @Test
    fun `should return another one when parked one car`() {
        val oneLot = parkingManager.findOneValidParkingLot()!!
            .also { it.park(Car("川A 11111")) }

        val anotherLot = parkingManager.findOneValidParkingLot()!!

        assertTrue(anotherLot in parkingLots)
        assertNotEquals(oneLot, anotherLot)
    }
}
